packages/synapse-react-client/src/components/HasAccess/HasAccessV2.tsx

```diff
@@ -104,64 +91,39 @@ function AccessIcon(props: { downloadType: FileHandleDownloadTypeEnum }) {
  *
  * To make download available, and determine if the file is downloadable via the web, see {@link DirectDownload.tsx}
  * @param entityId
- * @param entityVersionNumber
  * @returns
  */
-export function useGetFileHandleDownloadType(
+export function useGetRestrictionUiType(
   entityId: string,
-  entityVersionNumber?: string,
-): FileHandleDownloadTypeEnum | undefined {
+): RestrictionUiType | undefined {
   const { accessToken } = useSynapseContext()
-  const parsedVersionNumber = parseInt(entityVersionNumber ?? '')
-  const { data: entityBundle } = useGetEntityBundle(
-    entityId,
-    Number.isNaN(parsedVersionNumber) ? undefined : parsedVersionNumber,
-    {
-      includeEntity: true,
-      includePermissions: true,
-    },
-  )
-
-  const restrictionInformationRequest: RestrictionInformationRequest = useMemo(
-    () => ({
-      restrictableObjectType: RestrictableObjectType.ENTITY,
-      objectId: entityId,
-    }),
-    [entityId],
-  )
+  const isSignedIn = Boolean(accessToken)
 
-  const { data: restrictionInformation } = useGetRestrictionInformation(
-    restrictionInformationRequest,
-  )
+  const { data: restrictionInformation } = useGetRestrictionInformation({
```

Comment:
Get the restriction information (in SynapseTable, we'll prefetch this in a batch!)